pkg/scheduler/event_handler.go

```diff
@@ -164,10 +163,7 @@ func (s *Scheduler) updateCluster(oldObj, newObj interface{}) {
 		s.schedulerEstimatorWorker.Add(newCluster.Name)
 	}
 
-	switch {
-	case !equality.Semantic.DeepEqual(oldCluster.Labels, newCluster.Labels):
-		fallthrough
-	case !equality.Semantic.DeepEqual(oldCluster.Spec, newCluster.Spec):
+	if oldCluster.Generation != newCluster.Generation {
```

Comment:
If we are going to do this, make sure to add comments to explain the generation change probably caused by label change.